rotkehlchen/chain/ethereum/modules/liquity/trove.py

```diff
@@ -159,7 +181,11 @@ def get_positions(
                         f'Ignoring Liquity trove information. '
                         f'Failed to decode contract information. {e!s}.',
                     )
-        return data
+        final_result = GetPositionsResult(
```

Comment:
The `._calculate_total_collateral_ratio` can return `None` so you need to make it optional. So completely missing from the response if there is a failure.

What's more you need to add this into the api docs you edited.